### PR TITLE
[KRC_26] Adds OrderSummary component

### PR DESCRIPTION
**Pull request by mgolawski**
[KRC-26](https://digitalrig.atlassian.net/browse/KRC-26) OrderSummary component

# Description 
Screenshot 2019-04-01 14 29 47

## Checklist:

- `Text` component with option properties (isUnderline, isBold)
- `TextHighlight` composed from `Text` component
- `SectionHeader` component
- `OrderSummary` component`

### CODE REVIEW

File: src/components/Cart/__tests__/__snapshots__/Cart.test.js.snap [hunk 1]
```diff
@@ -85,10 +85,61 @@ Array [
   margin: 3rem 1.3rem;
```
Comment (blurbyte) [inline src/components/Cart/__tests__/__snapshots__/Cart.test.js.snap]: Mock OrderSummary in Cart too.

File: src/components/Cart/Cart.js [hunk 2]
```diff
@@ -3,13 +3,15 @@ import Item from '../Item';
...
 import NextStep from '../NextStep';
+import OrderSummary from '../OrderSummary';
 
 function Cart() {
   return (
     <>
       <section>
-        <h2>Cart placeholder</h2>
         <Item />
+        <hr />
```
Comment (blurbyte) [inline src/components/Cart/Cart.js]: Please avoid using **\<hr>**. Use styled border instead.

File: src/components/Cart/__tests__/__snapshots__/Cart.test.js.snap [hunk 3]
```diff
@@ -85,10 +85,61 @@ Array [
...
 }
 
+.c14 {
```
Comment (blurbyte) [inline src/components/Cart/__tests__/__snapshots__/Cart.test.js.snap]: Mock it here too.

File: src/components/OrderSummary/OrderSummary.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    <>
+      <SectionHeader title="Order Summary"> </SectionHeader>
+      <Wrapper>
+        <PriceRow>
+          <Text>
+            Subtotal<PriceBadge>$125.00</PriceBadge>
+          </Text>
+        </PriceRow>
+        <PriceRow>
+          <TextHighlight isBold>
```
Comment (blurbyte) [inline src/components/OrderSummary/OrderSummary.js]: Just use *bold*, without *is* (it's styled-components pattern).

File: src/components/OrderSummary/OrderSummary.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+            Kohl's Cash & Discounts<PriceBadge>-$18.75</PriceBadge>
+          </TextHighlight>
+        </PriceRow>
+        <PriceRow>
+          <Text isUnderline>
```
Comment (blurbyte) [inline src/components/OrderSummary/OrderSummary.js]: *underline* is enough.

File: src/components/OrderSummary/TotalPrice.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+import { fontSmall } from '../../styles/designTokens';
+import PriceBadge from './PriceBadge';
+import PriceRow from './PriceRow';
+
+const PriceSummary = styled.span`
+  font-size: ${fontSmall};
+  font-weight: bold;
+`;
+
+function TotalPrice() {
```
Comment (blurbyte) [inline src/components/OrderSummary/TotalPrice.js]: I think it is safe to pass as prop here.

File: src/components/Text/Text.js [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import styled from 'styled-components';
+import { fontSmall } from '../../styles/designTokens';
+
+const Text = styled.span`
+  font-size: ${fontSmall};
+  text-decoration: ${({ isUnderline }) => (isUnderline ? 'underline' : 'none')};
```
Comment (blurbyte) [inline src/components/Text/Text.js]: As per previous comments change to **underline** and **bold** (I know it is good idea to prefix flags with 'is' but in styled-components it is common pattern to not do it).

**Review by blurbyte — CHANGES_REQUESTED** [review]
Looks good. Please reduce size of snapshots by mocking OrderSummary. Don't prefix props from styled-components with 'is'.

File: src/components/Divider/index.js [hunk 8]
```diff
@@ -0,0 +1,3 @@
+import Divider from './Divider';
```
Comment (blurbyte) [inline src/components/Divider/index.js]: Now I understand :) But not really what I thought. I was rather thinking to add border to TotalPrice.

**Review by blurbyte — CHANGES_REQUESTED** [review]

File: src/components/Divider/index.js (hunk 8 above)
Comment (mgolawski) [reply src/components/Divider/index.js]: Divider is used in multiple places in the app. TotalPrice is one of 10.

**Review by blurbyte — APPROVED** [review]
Good job!

**Pull request merged**